src/web/app/pages-instructor/instructor-sessions-page/instructor-sessions-page.component.ts

```diff
@@ -197,7 +197,7 @@ export class InstructorSessionsPageComponent extends InstructorSessionModalPageC
                 'The feedback session has been copied. Please modify settings/questions as necessary.',
                 { courseid: createdFeedbackSession.courseId, fsname: createdFeedbackSession.feedbackSessionName });
           }, (resp: ErrorMessageOutput) => { this.statusMessageService.showErrorToast(resp.error.message); });
-    });
+    }).catch(() => this.isCopyOtherSessionLoading = false);
```

Comment:
I tried several ways and it seems to be no. I think they are for different purpose? The finalize is to set the boolean to false when it terminates on complete or errors and the catch is for any uncaught promise. Do you have any detailed idea to combine them?

Reply:
You could try converting the promise to an observable, and use `finalize` to catch them together

Reply:
I tried converting the promise to an observable 
```
from(modalRef.result.then((result: CopyFromOtherSessionsResult) => {
      this.copyFeedbackSession(result.fromFeedbackSession, result.newFeedbackSessionName, result.copyToCourseId)
          .subscribe((createdFeedbackSession: FeedbackSession) => {
            this.navigationService.navigateWithSuccessMessage(this.router, '/web/instructor/sessions/edit',
                'The feedback session has been copied. Please modify settings/questions as necessary.',
                { courseid: createdFeedbackSession.courseId, fsname: createdFeedbackSession.feedbackSessionName });
          }, (resp: ErrorMessageOutput) => { this.statusMessageService.showErrorToast(resp.error.message); });
    })).pipe(finalize(() => this.isCopyOtherSessionLoading = false))
        .subscribe(() => {});
```
However, this will cause the spinner to stop prematurely when the backend is still copying the session. I think it may be due to that the Observable that is converted from the Promise is different from the Observable copying the session, so when the Observable converted from the Promise finishes, the Observable copying the session is still ongoing. I would like to know in detail on how to do it correctly.
![copy20210622_191552](https://user-images.githubusercontent.com/53338059/122915488-60560900-d38e-11eb-8c63-864171197e4c.gif)

Reply:
I think it's a similar case to the other issue here https://github.com/TEAMMATES/teammates/pull/10893